frontend/web/components/App.js

```diff
@@ -350,12 +351,27 @@ const App = class extends Component {
       return <div>{this.props.children}</div>
     }
     const announcementValue = Utils.getFlagsmithJSONValue('announcement', null)
+    const announcementPerPageDismissed = flagsmith.getTrait(
```

Comment:
No need to pollute app with this logic, can go in the component

Reply:
I want to keep them separated so we can show both at the same time, a general announcement and a page-specific announcement. Of course, I'll resume as much logic as possible, however, these components are very simple, anyway.